// afip-apis/dist/lib/services/personaServiceA10.soap.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PersonaServiceA10 = void 0;
const xml2js = __importStar(require("xml2js"));
const soap11_1 = require("./soap11");
const debug_1 = __importDefault(require("debug"));
const debug = (0, debug_1.default)("afip-apis:personaA10");
class PersonaServiceA10 extends soap11_1.SoapService {
    static get serviceId() {
        return "ws_sr_padron_a10";
    }
    static get produccionWSDL() {
        return "https://aws.afip.gov.ar/sr-padron/webservices/personaServiceA10?WSDL";
    }
    static get testWSDL() {
        return "https://awshomo.afip.gov.ar/sr-padron/webservices/personaServiceA10?WSDL";
    }
    constructor(wsdlUrl) {
        super(wsdlUrl);
        this.getPersona = (input, options, extraHeaders) => {
            debug("getPersona;");
            extraHeaders = extraHeaders || {};
            const builder = new xml2js.Builder({ headless: true });
            const envelope = {
                "soapenv:Envelope": {
                    $: {
                        "xmlns:soapenv": "http://schemas.xmlsoap.org/soap/envelope/",
                        "xmlns:a10": "http://a10.soap.ws.server.puc.sr/"
                    },
                    "soapenv:Header": undefined,
                    "soapenv:Body": {
                        "a10:getPersona": input
                    }
                }
            };
            const soapEnvelope = builder.buildObject(envelope);
            debug("getPersona;soapEnvelope: %s", soapEnvelope);
            return new Promise((resolve, reject) => {
                return this.invoke(soapEnvelope, undefined, extraHeaders)
                    .then(r => resolve(r["ns2:getPersonaResponse"]))
                    .catch(err => reject(err));
            });
        };
        this.dummy = (input, options, extraHeaders) => {
            debug("dummy;");
            extraHeaders = extraHeaders || {};
            const builder = new xml2js.Builder({ headless: true });
            const envelope = {
                "soapenv:Envelope": {
                    $: {
                        "xmlns:soapenv": "http://schemas.xmlsoap.org/soap/envelope/",
                        "xmlns:a10": "http://a10.soap.ws.server.puc.sr/"
                    },
                    "soapenv:Header": undefined,
                    "soapenv:Body": {
                        "a10:dummy": undefined,
                    }
                }
            };
            const soapEnvelope = builder.buildObject(envelope);
            debug("dummy;soapEnvelope: %s", soapEnvelope);
            return new Promise((resolve, reject) => {
                return this.invoke(soapEnvelope, undefined, extraHeaders)
                    .then(r => resolve(r["ns2:dummyResponse"]))
                    .catch(err => reject(err));
            });
        };
        console.warn("ws_sr_padron_a10 fue eliminado de producción");
    }
}
exports.PersonaServiceA10 = PersonaServiceA10;
