// afip-apis/dist/lib/services/personaServiceA13.soap.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PersonaServiceA13 = void 0;
const xml2js = __importStar(require("xml2js"));
const soap11_1 = require("./soap11");
const debug_1 = __importDefault(require("debug"));
const debug = (0, debug_1.default)("afip-apis:personaA13");
const envelope = {
    "soapenv:Envelope": {
        $: {
            "xmlns:soapenv": "http://schemas.xmlsoap.org/soap/envelope/",
            "xmlns:a13": "http://a13.soap.ws.server.puc.sr/"
        },
        "soapenv:Header": undefined,
        "soapenv:Body": undefined,
    }
};
class PersonaServiceA13 extends soap11_1.SoapService {
    constructor() {
        super(...arguments);
        this.dummy = (input, options, extraHeaders) => {
            extraHeaders = extraHeaders || {};
            const builder = new xml2js.Builder({ headless: true });
            envelope["soapenv:Envelope"]["soapenv:Body"] = { "a13:dummy": input };
            const soapEnvelope = builder.buildObject(envelope);
            debug("dummy;soapEnvelope: %s", soapEnvelope);
            return new Promise((resolve, reject) => {
                return this.invoke(soapEnvelope, undefined, extraHeaders)
                    .then(r => resolve(r["ns2:dummyResponse"]))
                    .catch(err => {
                    return reject(err);
                });
            });
        };
        this.getIdPersonaListByDocumento = (input, options, extraHeaders) => {
            extraHeaders = extraHeaders || {};
            const builder = new xml2js.Builder({ headless: true });
            envelope["soapenv:Envelope"]["soapenv:Body"] = { "a13:getIdPersonaListByDocumento": input };
            const soapEnvelope = builder.buildObject(envelope);
            debug("getIdPersonaListByDocumento;soapEnvelope: %s", soapEnvelope);
            return new Promise((resolve, reject) => {
                return this.invoke(soapEnvelope, undefined, extraHeaders)
                    .then(r => {
                    return resolve(r["ns2:getIdPersonaListByDocumentoResponse"]);
                })
                    .catch(err => {
                    return reject(err);
                });
            });
        };
        this.getPersona = (input, options, extraHeaders) => {
            extraHeaders = extraHeaders || {};
            const builder = new xml2js.Builder({ headless: true });
            envelope["soapenv:Envelope"]["soapenv:Body"] = { "a13:getPersona": input };
            const soapEnvelope = builder.buildObject(envelope);
            debug("getPersona;soapEnvelope: %s", soapEnvelope);
            return new Promise((resolve, reject) => {
                return this.invoke(soapEnvelope, undefined, extraHeaders)
                    .then(r => {
                    return resolve(r["ns2:getPersonaResponse"]);
                })
                    .catch(err => {
                    return reject(err);
                });
            });
        };
    }
    static get serviceId() {
        return "ws_sr_padron_a13";
    }
    static get produccionWSDL() {
        return "https://aws.afip.gov.ar/sr-padron/webservices/personaServiceA13?WSDL";
    }
    static get testWSDL() {
        return "https://awshomo.afip.gov.ar/sr-padron/webservices/personaServiceA13?WSDL";
    }
}
exports.PersonaServiceA13 = PersonaServiceA13;
