// afip-apis/dist/lib/services/wsaa.soap.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LoginCmsSoap = void 0;
const error_server_class_1 = require("../shared/error.server.class");
const soap11_1 = require("./soap11");
const debug_1 = __importDefault(require("debug"));
const debug = (0, debug_1.default)("afip-apis:wsaa");
function tagNameProcessor(name) {
    return name.substring(0, 5).toLocaleLowerCase() === "soap:" ? name.substring(5, name.length).toLocaleLowerCase() : name;
}
const defaultParserOptions = {
    explicitRoot: false, explicitArray: false, ignoreAttrs: true,
    tagNameProcessors: [tagNameProcessor],
};
class LoginCmsSoap extends soap11_1.SoapService {
    constructor() {
        super(...arguments);
        this.loginCms = (input, options, extraHeaders) => {
            extraHeaders = extraHeaders || {};
            extraHeaders.SOAPAction = this.url;
            extraHeaders["Content-Type"] = "application/xml";
            if (!this.url) {
                return Promise.reject(new error_server_class_1.BaseError(409, "url-is-undefined", "La url no puede ser nula", undefined));
            }
            const soapEnvelope = `<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:wsaa="http://wsaa.view.sua.dvadac.desein.afip.gov">
        <soapenv:Header/>
        <soapenv:Body>
          <wsaa:loginCms>
            <wsaa:in0> ${input.in0}</wsaa:in0>
          </wsaa:loginCms>
        </soapenv:Body>
        </soapenv:Envelope>`;
            return new Promise((resolve, reject) => {
                return this.invoke(soapEnvelope, undefined, extraHeaders)
                    .then(r => {
                    return resolve(r.loginCmsResponse);
                })
                    .catch(err => {
                    return reject(err);
                });
            });
        };
    }
    static get produccionWSDL() {
        return "https://wsaa.afip.gov.ar/ws/services/LoginCms?WSDL";
    }
    static get testWSDL() {
        return "https://wsaahomo.afip.gov.ar/ws/services/LoginCms?WSDL";
    }
}
exports.LoginCmsSoap = LoginCmsSoap;
