// afip-apis/dist/lib/services/wsfexv1.soap.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Wsfexv1 = void 0;
const xml2js = __importStar(require("xml2js"));
const error_server_class_1 = require("../shared/error.server.class");
const soap12_1 = require("./soap12");
const xmlns = "http://ar.gov.afip.dif.fexv1/";
class Wsfexv1 extends soap12_1.SoapService {
    constructor() {
        super(...arguments);
        this.FEXAuthorize = (input, options, extraHeaders) => {
            const builder = new xml2js.Builder({ headless: true });
            extraHeaders = extraHeaders || {};
            extraHeaders.SOAPAction = "http://ar.gov.afip.dif.fexv1/FEXAuthorize";
            const envelope = {
                "soap:Envelope": {
                    $: {
                        "xmlns:soap": "http://www.w3.org/2003/05/soap-envelope"
                    },
                    "soap:Header": undefined,
                    "soap:Body": {
                        $: { xmlns: xmlns },
                        FEXAuthorize: input
                    }
                }
            };
            const soapEnvelope = builder.buildObject(envelope);
            return new Promise((resolve, reject) => {
                return this.invoke(soapEnvelope, undefined, extraHeaders)
                    .then(r => {
                    if (!r) {
                        return this.emptyResponse();
                    }
                    return resolve(r.FEXAuthorizeResponse);
                })
                    .catch(err => reject(err));
            });
        };
        this.FEXCheck_Permiso = (input, options, extraHeaders) => {
            const builder = new xml2js.Builder({ headless: true });
            extraHeaders = extraHeaders || {};
            extraHeaders.SOAPAction = "http://ar.gov.afip.dif.fexv1/FEXCheck_Permiso";
            const envelope = {
                "soap:Envelope": {
                    $: {
                        "xmlns:soap": "http://www.w3.org/2003/05/soap-envelope"
                    },
                    "soap:Header": undefined,
                    "soap:Body": {
                        $: { xmlns: xmlns },
                        FEXCheck_Permiso: input
                    }
                }
            };
            const soapEnvelope = builder.buildObject(envelope);
            return new Promise((resolve, reject) => {
                return this.invoke(soapEnvelope, undefined, extraHeaders)
                    .then(r => {
                    if (!r) {
                        return this.emptyResponse();
                    }
                    return resolve(r.FEXCheck_PermisoResponse);
                })
                    .catch(err => reject(err));
            });
        };
        this.FEXDummy = (options, extraHeaders) => {
            const builder = new xml2js.Builder({ headless: true });
            extraHeaders = extraHeaders || {};
            extraHeaders.SOAPAction = "http://ar.gov.afip.dif.fexv1/FEXDummy";
            const envelope = {
                "soap:Envelope": {
                    $: {
                        "xmlns:soap": "http://www.w3.org/2003/05/soap-envelope"
                    },
                    "soap:Header": undefined,
                    "soap:Body": {
                        $: { xmlns: xmlns }
                    }
                }
            };
            const soapEnvelope = builder.buildObject(envelope);
            return new Promise((resolve, reject) => {
                return this.invoke(soapEnvelope, undefined, extraHeaders)
                    .then(r => {
                    if (!r) {
                        return this.emptyResponse();
                    }
                    return resolve(r.FEXDummyResponse);
                })
                    .catch(err => reject(err));
            });
        };
        this.FEXGetCMP = (input, options, extraHeaders) => {
            const builder = new xml2js.Builder({ headless: true });
            extraHeaders = extraHeaders || {};
            extraHeaders.SOAPAction = "http://ar.gov.afip.dif.fexv1/FEXGetCMP";
            const envelope = {
                "soap:Envelope": {
                    $: {
                        "xmlns:soap": "http://www.w3.org/2003/05/soap-envelope"
                    },
                    "soap:Header": undefined,
                    "soap:Body": {
                        $: { xmlns: xmlns },
                        FEXGetCMP: input
                    }
                }
            };
            const soapEnvelope = builder.buildObject(envelope);
            return new Promise((resolve, reject) => {
                return this.invoke(soapEnvelope, undefined, extraHeaders)
                    .then(r => {
                    if (!r) {
                        return this.emptyResponse();
                    }
                    return resolve(r.FEXGetCMPResponse);
                })
                    .catch(err => reject(err));
            });
        };
        this.FEXGetLast_CMP = (input, options, extraHeaders) => {
            const builder = new xml2js.Builder({ headless: true });
            extraHeaders = extraHeaders || {};
            extraHeaders.SOAPAction = "http://ar.gov.afip.dif.fexv1/FEXGetLast_CMP";
            const envelope = {
                "soap:Envelope": {
                    $: {
                        "xmlns:soap": "http://www.w3.org/2003/05/soap-envelope"
                    },
                    "soap:Header": undefined,
                    "soap:Body": {
                        $: { xmlns: xmlns },
                        FEXGetLast_CMP: input
                    }
                }
            };
            const soapEnvelope = builder.buildObject(envelope);
            return new Promise((resolve, reject) => {
                return this.invoke(soapEnvelope, undefined, extraHeaders)
                    .then(r => {
                    if (!r) {
                        return this.emptyResponse();
                    }
                    return resolve(r.FEXGetLast_CMPResponse);
                })
                    .catch(err => reject(err));
            });
        };
        this.FEXGetLast_ID = (input, options, extraHeaders) => {
            const builder = new xml2js.Builder({ headless: true });
            extraHeaders = extraHeaders || {};
            extraHeaders.SOAPAction = "http://ar.gov.afip.dif.fexv1/FEXGetLast_ID";
            const envelope = {
                "soap:Envelope": {
                    $: {
                        "xmlns:soap": "http://www.w3.org/2003/05/soap-envelope"
                    },
                    "soap:Header": undefined,
                    "soap:Body": {
                        $: { xmlns: xmlns },
                        FEXGetLast_ID: input
                    }
                }
            };
            const soapEnvelope = builder.buildObject(envelope);
            return new Promise((resolve, reject) => {
                return this.invoke(soapEnvelope, undefined, extraHeaders)
                    .then(r => {
                    if (!r) {
                        return this.emptyResponse();
                    }
                    return resolve(r.FEXGetLast_IDResponse);
                })
                    .catch(err => reject(err));
            });
        };
        this.FEXGetPARAM_Cbte_Tipo = (input, options, extraHeaders) => {
            const builder = new xml2js.Builder({ headless: true });
            extraHeaders = extraHeaders || {};
            extraHeaders.SOAPAction = "http://ar.gov.afip.dif.fexv1/FEXGetPARAM_Cbte_Tipo";
            const envelope = {
                "soap:Envelope": {
                    $: {
                        "xmlns:soap": "http://www.w3.org/2003/05/soap-envelope"
                    },
                    "soap:Header": undefined,
                    "soap:Body": {
                        $: { xmlns: xmlns },
                        FEXGetPARAM_Cbte_Tipo: input
                    }
                }
            };
            const soapEnvelope = builder.buildObject(envelope);
            return new Promise((resolve, reject) => {
                return this.invoke(soapEnvelope, undefined, extraHeaders)
                    .then(r => {
                    if (!r) {
                        return this.emptyResponse();
                    }
                    return resolve(r.FEXGetPARAM_Cbte_TipoResponse);
                })
                    .catch(err => reject(err));
            });
        };
        this.FEXGetPARAM_Ctz = (input, options, extraHeaders) => {
            const builder = new xml2js.Builder({ headless: true });
            extraHeaders = extraHeaders || {};
            extraHeaders.SOAPAction = "http://ar.gov.afip.dif.fexv1/FEXGetPARAM_Ctz";
            const envelope = {
                "soap:Envelope": {
                    $: {
                        "xmlns:soap": "http://www.w3.org/2003/05/soap-envelope"
                    },
                    "soap:Header": undefined,
                    "soap:Body": {
                        $: { xmlns: xmlns },
                        FEXGetPARAM_Ctz: input
                    }
                }
            };
            const soapEnvelope = builder.buildObject(envelope);
            return new Promise((resolve, reject) => {
                return this.invoke(soapEnvelope, undefined, extraHeaders)
                    .then(r => {
                    if (!r) {
                        return this.emptyResponse();
                    }
                    return resolve(r.FEXGetPARAM_CtzResponse);
                })
                    .catch(err => reject(err));
            });
        };
        this.FEXGetPARAM_DST_CUIT = (input, options, extraHeaders) => {
            const builder = new xml2js.Builder({ headless: true });
            extraHeaders = extraHeaders || {};
            extraHeaders.SOAPAction = "http://ar.gov.afip.dif.fexv1/FEXGetPARAM_DST_CUIT";
            const envelope = {
                "soap:Envelope": {
                    $: {
                        "xmlns:soap": "http://www.w3.org/2003/05/soap-envelope"
                    },
                    "soap:Header": undefined,
                    "soap:Body": {
                        $: { xmlns: xmlns },
                        FEXGetPARAM_DST_CUIT: input
                    }
                }
            };
            const soapEnvelope = builder.buildObject(envelope);
            return new Promise((resolve, reject) => {
                return this.invoke(soapEnvelope, undefined, extraHeaders)
                    .then(r => {
                    if (!r) {
                        return this.emptyResponse();
                    }
                    return resolve(r.FEXGetPARAM_DST_CUITResponse);
                })
                    .catch(err => reject(err));
            });
        };
        this.FEXGetPARAM_DST_pais = (input, options, extraHeaders) => {
            const builder = new xml2js.Builder({ headless: true });
            extraHeaders = extraHeaders || {};
            extraHeaders.SOAPAction = "http://ar.gov.afip.dif.fexv1/FEXGetPARAM_DST_pais";
            const envelope = {
                "soap:Envelope": {
                    $: {
                        "xmlns:soap": "http://www.w3.org/2003/05/soap-envelope"
                    },
                    "soap:Header": undefined,
                    "soap:Body": {
                        $: { xmlns: xmlns },
                        FEXGetPARAM_DST_pais: input
                    }
                }
            };
            const soapEnvelope = builder.buildObject(envelope);
            return new Promise((resolve, reject) => {
                return this.invoke(soapEnvelope, undefined, extraHeaders)
                    .then(r => {
                    if (!r) {
                        return this.emptyResponse();
                    }
                    return resolve(r.FEXGetPARAM_DST_paisResponse);
                })
                    .catch(err => reject(err));
            });
        };
        this.FEXGetPARAM_Idiomas = (input, options, extraHeaders) => {
            const builder = new xml2js.Builder({ headless: true });
            extraHeaders = extraHeaders || {};
            extraHeaders.SOAPAction = "http://ar.gov.afip.dif.fexv1/FEXGetPARAM_Idiomas";
            const envelope = {
                "soap:Envelope": {
                    $: {
                        "xmlns:soap": "http://www.w3.org/2003/05/soap-envelope"
                    },
                    "soap:Header": undefined,
                    "soap:Body": {
                        $: { xmlns: xmlns },
                        FEXGetPARAM_Idiomas: input
                    }
                }
            };
            const soapEnvelope = builder.buildObject(envelope);
            return new Promise((resolve, reject) => {
                return this.invoke(soapEnvelope, undefined, extraHeaders)
                    .then(r => {
                    if (!r) {
                        return this.emptyResponse();
                    }
                    return resolve(r.FEXGetPARAM_IdiomasResponse);
                })
                    .catch(err => reject(err));
            });
        };
        this.FEXGetPARAM_Incoterms = (input, options, extraHeaders) => {
            const builder = new xml2js.Builder({ headless: true });
            extraHeaders = extraHeaders || {};
            extraHeaders.SOAPAction = "http://ar.gov.afip.dif.fexv1/FEXGetPARAM_Incoterms";
            const envelope = {
                "soap:Envelope": {
                    $: {
                        "xmlns:soap": "http://www.w3.org/2003/05/soap-envelope"
                    },
                    "soap:Header": undefined,
                    "soap:Body": {
                        $: { xmlns: xmlns },
                        FEXGetPARAM_Incoterms: input
                    }
                }
            };
            const soapEnvelope = builder.buildObject(envelope);
            return new Promise((resolve, reject) => {
                return this.invoke(soapEnvelope, undefined, extraHeaders)
                    .then(r => {
                    if (!r) {
                        return this.emptyResponse();
                    }
                    return resolve(r.FEXGetPARAM_IncotermsResponse);
                })
                    .catch(err => reject(err));
            });
        };
        this.FEXGetPARAM_MON = (input, options, extraHeaders) => {
            const builder = new xml2js.Builder({ headless: true });
            extraHeaders = extraHeaders || {};
            extraHeaders.SOAPAction = "http://ar.gov.afip.dif.fexv1/FEXGetPARAM_MON";
            const envelope = {
                "soap:Envelope": {
                    $: {
                        "xmlns:soap": "http://www.w3.org/2003/05/soap-envelope"
                    },
                    "soap:Header": undefined,
                    "soap:Body": {
                        $: { xmlns: xmlns },
                        FEXGetPARAM_MON: input
                    }
                }
            };
            const soapEnvelope = builder.buildObject(envelope);
            return new Promise((resolve, reject) => {
                return this.invoke(soapEnvelope, undefined, extraHeaders)
                    .then(r => {
                    if (!r) {
                        return this.emptyResponse();
                    }
                    return resolve(r.FEXGetPARAM_MONResponse);
                })
                    .catch(err => reject(err));
            });
        };
        this.FEXGetPARAM_MON_CON_COTIZACION = (input, options, extraHeaders) => {
            const builder = new xml2js.Builder({ headless: true });
            extraHeaders = extraHeaders || {};
            extraHeaders.SOAPAction = "http://ar.gov.afip.dif.fexv1/FEXGetPARAM_MON_CON_COTIZACION";
            const envelope = {
                "soap:Envelope": {
                    $: {
                        "xmlns:soap": "http://www.w3.org/2003/05/soap-envelope"
                    },
                    "soap:Header": undefined,
                    "soap:Body": {
                        $: { xmlns: xmlns },
                        FEXGetPARAM_MON_CON_COTIZACION: input
                    }
                }
            };
            const soapEnvelope = builder.buildObject(envelope);
            return new Promise((resolve, reject) => {
                return this.invoke(soapEnvelope, undefined, extraHeaders)
                    .then(r => {
                    if (!r) {
                        return this.emptyResponse();
                    }
                    return resolve(r.FEXGetPARAM_MON_CON_COTIZACIONResponse);
                })
                    .catch(err => reject(err));
            });
        };
        this.FEXGetPARAM_Opcionales = (input, options, extraHeaders) => {
            const builder = new xml2js.Builder({ headless: true });
            extraHeaders = extraHeaders || {};
            extraHeaders.SOAPAction = "http://ar.gov.afip.dif.fexv1/FEXGetPARAM_Opcionales";
            const envelope = {
                "soap:Envelope": {
                    $: {
                        "xmlns:soap": "http://www.w3.org/2003/05/soap-envelope"
                    },
                    "soap:Header": undefined,
                    "soap:Body": {
                        $: { xmlns: xmlns },
                        FEXGetPARAM_Opcionales: input
                    }
                }
            };
            const soapEnvelope = builder.buildObject(envelope);
            return new Promise((resolve, reject) => {
                return this.invoke(soapEnvelope, undefined, extraHeaders)
                    .then(r => {
                    if (!r) {
                        return this.emptyResponse();
                    }
                    return resolve(r.FEXGetPARAM_OpcionalesResponse);
                })
                    .catch(err => reject(err));
            });
        };
        this.FEXGetPARAM_PtoVenta = (input, options, extraHeaders) => {
            const builder = new xml2js.Builder({ headless: true });
            extraHeaders = extraHeaders || {};
            extraHeaders.SOAPAction = "http://ar.gov.afip.dif.fexv1/FEXGetPARAM_PtoVenta";
            const envelope = {
                "soap:Envelope": {
                    $: {
                        "xmlns:soap": "http://www.w3.org/2003/05/soap-envelope"
                    },
                    "soap:Header": undefined,
                    "soap:Body": {
                        $: { xmlns: xmlns },
                        FEXGetPARAM_PtoVenta: input
                    }
                }
            };
            const soapEnvelope = builder.buildObject(envelope);
            return new Promise((resolve, reject) => {
                return this.invoke(soapEnvelope, undefined, extraHeaders)
                    .then(r => {
                    if (!r) {
                        return this.emptyResponse();
                    }
                    return resolve(r.FEXGetPARAM_PtoVentaResponse);
                })
                    .catch(err => reject(err));
            });
        };
        this.FEXGetPARAM_Tipo_Expo = (input, options, extraHeaders) => {
            const builder = new xml2js.Builder({ headless: true });
            extraHeaders = extraHeaders || {};
            extraHeaders.SOAPAction = "http://ar.gov.afip.dif.fexv1/FEXGetPARAM_Tipo_Expo";
            const envelope = {
                "soap:Envelope": {
                    $: {
                        "xmlns:soap": "http://www.w3.org/2003/05/soap-envelope"
                    },
                    "soap:Header": undefined,
                    "soap:Body": {
                        $: { xmlns: xmlns },
                        FEXGetPARAM_Tipo_Expo: input
                    }
                }
            };
            const soapEnvelope = builder.buildObject(envelope);
            return new Promise((resolve, reject) => {
                return this.invoke(soapEnvelope, undefined, extraHeaders)
                    .then(r => {
                    if (!r) {
                        return this.emptyResponse();
                    }
                    return resolve(r.FEXGetPARAM_Tipo_ExpoResponse);
                })
                    .catch(err => reject(err));
            });
        };
        this.FEXGetPARAM_UMed = (input, options, extraHeaders) => {
            const builder = new xml2js.Builder({ headless: true });
            extraHeaders = extraHeaders || {};
            extraHeaders.SOAPAction = "http://ar.gov.afip.dif.fexv1/FEXGetPARAM_UMed";
            const envelope = {
                "soap:Envelope": {
                    $: {
                        "xmlns:soap": "http://www.w3.org/2003/05/soap-envelope"
                    },
                    "soap:Header": undefined,
                    "soap:Body": {
                        $: { xmlns: xmlns },
                        FEXGetPARAM_UMed: input
                    }
                }
            };
            const soapEnvelope = builder.buildObject(envelope);
            return new Promise((resolve, reject) => {
                return this.invoke(soapEnvelope, undefined, extraHeaders)
                    .then(r => {
                    if (!r) {
                        return this.emptyResponse();
                    }
                    return resolve(r.FEXGetPARAM_UMedResponse);
                })
                    .catch(err => reject(err));
            });
        };
    }
    static get serviceId() {
        return "wsfex";
    }
    static get produccionWSDL() {
        return "https://servicios1.afip.gov.ar/wsfexv1/service.asmx?WSDL";
    }
    static get testWSDL() {
        return "https://wswhomo.afip.gov.ar/wsfexv1/service.asmx?WSDL";
    }
    emptyResponse() {
        return Promise.reject(new error_server_class_1.BaseError(500, "empty-response", "La respuesta del webservice esta vacia", undefined));
    }
}
exports.Wsfexv1 = Wsfexv1;
