// afip-apis/dist/test/test-wslpg.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const cache_login_1 = require("./cache-login");
const afip_apis_1 = require("../afip-apis");
const DEFAULT_CERTIFICATE = "./private/certificate/TEST/12.2023/certificate.crt";
const DEFAULT_CERTIFICATE_KEY = "./private/certificate/TEST/12.2023/private.key";
const DEFAULT_URLWSAAWSDL = "https://wsaahomo.afip.gov.ar/ws/services/LoginCms?WSDL";
const cacheLogin = cache_login_1.CacheLogin.Instance;
cacheLogin.certificateKey = DEFAULT_CERTIFICATE_KEY;
cacheLogin.certificatePath = DEFAULT_CERTIFICATE;
cacheLogin.wsaawsdl = DEFAULT_URLWSAAWSDL;
cacheLogin.getTicket(afip_apis_1.WsLpgv120.serviceId)
    .then(ticket => {
    console.log("=== ticket ===");
    console.log(JSON.stringify(ticket));
    const wsLpgv120 = new afip_apis_1.WsLpgv120(afip_apis_1.WsLpgv120.testWSDL);
    return wsLpgv120.dummy()
        .then(r => {
        console.log(`===dummy===\n${JSON.stringify(r)}`);
        return wsLpgv120.liquidacionXNroOrdenConsultar({
            auth: {
                token: ticket.credentials.token,
                sign: ticket.credentials.sign,
                cuit: 20220536999
            },
            ptoEmision: 1,
            nroOrden: 1
        });
    })
        .then(r => {
        console.log(`===liquidacionXNroOrdenConsultar===\n${JSON.stringify(r)}`);
        return wsLpgv120.provinciasConsultar({
            auth: {
                token: ticket.credentials.token,
                sign: ticket.credentials.sign,
                cuit: 20220536999
            }
        });
    })
        .then(r => {
        console.log(`===provinciasConsultar===\n${JSON.stringify(r)}`);
        return wsLpgv120.tipoDeduccionConsultar({
            auth: {
                token: ticket.credentials.token,
                sign: ticket.credentials.sign,
                cuit: 20220536999
            }
        });
    })
        .then(r => {
        console.log(`===tipoDeduccionConsultar===\n${JSON.stringify(r)}`);
    });
})
    .then(() => {
    console.log("=== FIN ===");
})
    .catch(err => {
    console.error(err);
});
